makefiles/mcuboot.mk

```diff
@@ -1,49 +1,52 @@
+.PHONY: mcuboot mcuboot-create-key mcuboot-flash-bootloader mcuboot-flash
+
 ifdef MCUBOOT_SLOT0_SIZE
 
 IMGTOOL ?= $(RIOTTOOLS)/mcuboot/imgtool.py
-override IMGTOOL := $(abspath $(IMGTOOL))
+IMGTOOL_KEYFLAGS ?= keygen -t rsa-2048
 
-BINFILE ?= $(BINDIR)/$(APPLICATION).bin
 SIGN_BINFILE = $(BINDIR)/signed-$(APPLICATION).bin
 MCUBOOT_KEYFILE ?= $(BINDIR)/key.pem
-MCUBOOT_BIN ?= $(BINDIR)/mcuboot.bin
-MCUBOOT_BIN_URL ?= http://download.riot-os.org/mynewt.mcuboot.bin
-MCUBOOT_BIN_MD5 ?= 0c71a0589bd3709fc2d90f07a0035ce7
 
-export IMAGE_HDR_SIZE ?= 512
+MCUBOOT_ELF ?= $(ELFFILE:.elf=.mcuboot.elf)
+MCUBOOT_BIN ?= $(MCUBOOT_ELF:.elf=.bin)
+
+MCUBOOT_LOADER_BIN ?= $(BINDIR)/mcuboot.bin
```

Comment:
I prefered `MCUBOOT_BIN` has it is the `mcuboot` binary file. For me it was completely clear. Here `loader` does not mean anything to me, at least `bootloader` would be clear.

However renaming the file to keep the full `mynewt.mcuboot.bin` name would prevent possible issues with the filename in `bin`.